### PR TITLE
DOC: df.to_html documentation incorrectly contains min_rows optional param

### LINKED ISSUE #44304: DOC: df.to_html documentation incorrectly contains min_rows optional param
### 

- [X] I have checked that the issue still exists on the latest versions of the docs on `master` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_html.html

### Documentation problem

The documentation contains an optional parameter `min_rows`, but the source code shows that no option is supported. See source code [here](https://github.com/pandas-dev/pandas/blob/v1.3.4/pandas/core/frame.py#L2711).

Verify this optional param is not supported with the following code:
```
import pandas as pd
df = pd.DataFrame({'data': [1, 2, 3]})
df.to_html(max_rows=2, min_rows=1)
```
Which outputs the error
```
      1 import pandas as pd
      2 df = pd.DataFrame({'data': [1, 2, 3]})
----> 3 df.to_html(max_rows=2, min_rows=1)

TypeError: to_html() got an unexpected keyword argument 'min_rows'
```

### Suggested fix for documentation

Remove the `min_rows` optional param from this documentation.

### PR DESCRIPTION
**Pull request by loicdiridollou**
- [x] closes #44304
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [NA] whatsnew entry

Moved min_rows parameter out of the common docstring that is used by both DataFrame.to_html and DataFrame.to_string and reintroduced the parameter in DataFrame.to_string only

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1078,6 +1078,9 @@ def to_string(
         %(shared_params)s
         line_width : int, optional
             Width to wrap a line in characters.
+        min_rows : int, optional
```
Comment (jreback) [inline pandas/core/frame.py:1081]: why did you move the parmeter?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 1 above)
Comment (loicdiridollou) [reply pandas/core/frame.py:1081]: The parameter `min_rows` used to be documented in `io.formats.frame.py` using a common docstring shared for `DataFrame.to_html` and `DataFrame.to_string`.
However the parameter is only used in `DataFrame.to_string` so that did not make sense to have it in a shared docstring (thus the confusion that led to the creation of issue #44304).
So I moved the documentation of `min_rows` out of the common docstring and added it only for `DataFrame.to_string` method (so it does not appear for `DataFrame.to_html` once we rebuild the documentation).

**Review by phofl — APPROVED** [review]
lgtm

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @loicdiridollou

**Comment by simonjayhawkins** [comment]
reordering parameters that are not keyword only is technically a breaking change even though it is unlikely that users would pass all parameters by position.

Maybe needs a release note.
